app/scripts/migrations/094.ts

```diff
@@ -0,0 +1,89 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty, isObject } from '@metamask/utils';
+import { NetworkStatus } from '@metamask/network-controller';
+import { NetworkType } from '@metamask/controller-utils';
+
+export const version = 94;
+
+/**
+ * Migrate NetworkDetails & NetworkStatus state on the NetworkConroller to NetworksMetadata and add selectedNetworkClientId
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(originalVersionedData: {
+  meta: { version: number };
+  data: Record<string, unknown>;
+}) {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  versionedData.data = transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
+  if (
+    hasProperty(state, 'NetworkController') &&
+    isObject(state.NetworkController) &&
+    hasProperty(state.NetworkController, 'providerConfig') &&
+    isObject(state.NetworkController.providerConfig) &&
+    (hasProperty(state.NetworkController.providerConfig, 'id') ||
+      (hasProperty(state.NetworkController.providerConfig, 'type') &&
+        state.NetworkController.providerConfig.type !== NetworkType.rpc))
+  ) {
+    const selectedNetworkClientId =
+      state.NetworkController.providerConfig.id ||
```

Comment:
Should the selectedNetworkId be derived from the providerConfig using the method we have for doing that?
```
 buildCustomNetworkClientId(
        providerConfig,
        this.state.networkConfigurations,
      );
```

Reply:
nvm just looked into the code more... mb!